orchestrator/cli/domain_gen_helpers/product_helpers.py

```diff
@@ -0,0 +1,204 @@
+
+    Args:
+        - create_block_relations: Dict with product names by product block
+            - key: product block name.
+            - value: Set of product names to relate with.
+
+    Returns: List of SQL strings to create subscription instances.
+    """
+
+    def create_block_relation(block_name: str, product_names: Set[str]) -> str:
+        block_id_sql = get_product_block_id(block_name)
+
+        def create_block_relation_dict(product_name: str) -> Dict[str, Query]:
+            product_id_sql = get_product_id(product_name)
+            return {"product_block_id": block_id_sql, "product_id": product_id_sql}
+
+        product_product_block_relation_dicts = [
+            create_block_relation_dict(product_name) for product_name in product_names
+        ]
+        return sql_compile(Insert(product_product_block_association).values(product_product_block_relation_dicts))
+
+    return [create_block_relation(*item) for item in create_block_relations.items()]
+
+
+def generate_create_product_instance_relations_sql(product_to_block_relations: Dict[str, Set[str]]) -> List[str]:
+    """Generate SQL to create subscription instances for existing subscriptions.
+
+    Args:
+        - product_to_block_relations: Dict with product blocks by resource type
+            - key: product block name.
+            - value: Set of product names to relate with.
+
+    Returns: List of SQL strings to create subscription instances.
+    """
+
+    def create_subscription_instance_relations(block_name: str, product_names: Set[str]) -> str:
+        product_block_id = get_product_block_id(block_name)
+
+        def map_subscription_instance_relations(product_name: str) -> List[Dict[str, Union[str, Query]]]:
+            product_id_sql = get_product_id(product_name)
+            subscription_ids = (
+                SubscriptionTable.query.where(SubscriptionTable.product_id.in_(product_id_sql))
+                .with_entities(SubscriptionTable.subscription_id)
+                .all()
+            )
+            if not subscription_ids:
+                return []
+            return [
+                {"subscription_id": subscription_id[0], "product_block_id": product_block_id}
+                for subscription_id in subscription_ids
+            ]
+
+        subscription_relation_dicts = list(
+            flatten([map_subscription_instance_relations(block_name) for block_name in product_names])
+        )
+        subscription_block_relations = ""
+        if subscription_relation_dicts:
+            subscription_block_relations = sql_compile(
+                Insert(SubscriptionInstanceTable).values(subscription_relation_dicts)
+            )
+        return subscription_block_relations
+
+    create_subscription_instance_sqls = [
+        create_subscription_instance_relations(*item) for item in product_to_block_relations.items()
+    ]
+    return [sub_instance_sql for sub_instance_sql in create_subscription_instance_sqls if sub_instance_sql]
```

Comment:
With help of yield and `from itertools import chain` this could be condensed a little bit ( I didn't check mypy though..)



```python
            for subscription_id in subscription_ids:
                yield {"subscription_id": subscription_id[0], "product_block_id": product_block_id}

        subscription_relation_dicts = list(
            chain.from_iterable(map_subscription_instance_relations(block_name) for block_name in product_names)
        )
        if subscription_relation_dicts:
            yield sql_compile(Insert(SubscriptionInstanceTable).values(subscription_relation_dicts))

    return list(
        chain.from_iterable(
            create_subscription_instance_relations(*item) for item in product_to_block_relations.items()
        )
    )
```

Optional though; depending on how much time you have left